Support for OpenSSL 1.0.1

Builds against older versions of OpenSSL fail due to usage of `SSL_CTX_set_ecdh_auto`

```
/home/jenkins/workspace/icinga2-snapshot/deb-debian-jessie-1binary/arch/x86/icinga2/lib/base/tlsutility.cpp: In function 'void icinga::SetupSslContext(SSL_CTX*, const icinga::String&, const icinga::String&, const icinga::String&)':
/home/jenkins/workspace/icinga2-snapshot/deb-debian-jessie-1binary/arch/x86/icinga2/lib/base/tlsutility.cpp:77:37: error: 'SSL_CTX_set_ecdh_auto' was not declared in this scope
  SSL_CTX_set_ecdh_auto(sslContext, 1);
```
* https://build.icinga.com/job/icinga2-snapshot/job/deb-debian-jessie-1binary/arch=x86/328/console
* https://build.icinga.com/job/icinga2-snapshot/job/rpm-centos-6-1binary/arch=x86_64/343/console

Happens on:

* RHEL/CentOS 6 `OpenSSL 1.0.1e-fips 11 Feb 2013`
* Debian jessie `OpenSSL 1.0.1t  3 May 2016`

--- a/lib/base/tlsutility.cpp
+++ b/lib/base/tlsutility.cpp
@@ -74,7 +74,9 @@ static void SetupSslContext(SSL_CTX *sslContext, const String& pubkey, const Str
 	SSL_CTX_set_session_id_context(sslContext, (const unsigned char *)"Icinga 2", 8);
 
 	// Explicitly load ECC ciphers, required on el7 - https://github.com/Icinga/icinga2/issues/7247
+#ifdef SSL_CTX_set_ecdh_auto
 	SSL_CTX_set_ecdh_auto(sslContext, 1);
+#endif /* SSL_CTX_set_ecdh_auto */
 
 	if (!pubkey.IsEmpty()) {
 		if (!SSL_CTX_use_certificate_chain_file(sslContext, pubkey.CStr())) {

